src/mono/mono/mini/mini-amd64.h

```diff
@@ -176,6 +176,10 @@ struct sigcontext {
  * reproduceable results for benchmarks */
 #define MONO_ARCH_CODE_ALIGNMENT 32
 
+#if defined(TARGET_OSX) || defined(TARGET_APPLE_MOBILE)
+#define MONO_ARCH_HAVE_SWIFTCALL 1
```

Comment:
I think that it could be set in `src/mono/CMakeLists.txt` by something like:
```
if ((TARGET_AMD64 OR TARGET_ARM64) AND (TARGET_OSX or TARGET_APPLE_MOBILE))
    set(MONO_ARCH_HAVE_SWIFTCALL 1)
endif()
```
However, CoreCLR also does set it in the respective arch files e.g. https://github.com/dotnet/runtime/blob/a5c1c9f364817a1fe2aa306ed33963cb41ab63fc/src/coreclr/jit/targetamd64.h#L563

What do you think is better? (cc: @kotlarmilos, @jkurdek)

Reply:
It might be cleaner to move it to `src/mono/CMakeLists.txt`, but don't have a strong opinion about it. Since this is not a new change, I suggest doing it in a follow-up PR.